overlord/snapstate/snapstate.go

```diff
@@ -1354,7 +1354,46 @@ func canDisable(si *snap.Info) bool {
 	return true
 }
 
+// baseInUse returns true if the given base is needed by another snap
+func baseInUse(st *state.State, base *snap.Info) bool {
+	snapStates, err := All(st)
+	if err != nil {
+		return false
+	}
+	for _, snapst := range snapStates {
+		if snapInfo, err := snapst.CurrentInfo(); err == nil {
```

Comment:
I think we should check the whole `snapst.Sequence` in case a previous revision uses that base, while the current one does not.